apps/storefront/next.config.js

```diff
@@ -49,16 +49,20 @@ module.exports = withBundleAnalyzer({
         source: "/checkout/(.*)",
         headers: [{ key: "x-frame-options", value: "ALLOWALL" }],
       },
+      /**
+       * TODO. Ensure this is valid approach. Best if it can be limited to Saleor,
+       *   so maybe it should use NEXT_PUBLIC_API_URI?
+       */
+      {
+        source: "/dashboard/(.*)",
+        headers: [{ key: "content-security-policy", value: "frame-ancestors *" }],
+      },
     ];
   },
   async rewrites() {
     const cloudDeploymentUrl = process.env.CLOUD_DEPLOYMENT_URL;
 
     return [
-      {
-        source: "/checkout/",
-        destination: `${process.env.NEXT_PUBLIC_CHECKOUT_URL}/`,
```

Comment:
Can you explain why? I looked on the codebase and I replaced a single instance of this redirect here:
https://github.com/saleor/react-storefront/pull/592/files#diff-e69091281753d50627a3bc1999dbb8e8abde136c55a1d841a299431142d00077L29

(at least I haven't found more)

This operation (fetching redirect URL) can be cached and is almost one-time, when user wants to redirect to the checkout. I don't see any performance issues

Reply:
I see. Middlewares sound good. Still I think it will only need to fetch this URL once, for this single call, right?

We can even fetch redirect URL on frontend and then redirect to `/checkout?checkoutURL=${fetchedCheckoutUrl}&checkoutId=${someID}`, which will be intercepted by middleware and rewritten?

Reply:
Each request to `/checkout` will trigger the middleware because each od them needs to be rewritten.
You can try but I think every script, static file, all medias, and api calls will go through it and it'll be slow.

Reply:
What are other options?

Reply:
Rethinking our auth flow for cross domain requests.